File: backend/proposal/core/test/order_test.py
import unittest
import time
from core.order_item import OrderItem
from core.product import Product
from core.order import Order
from core.customer import Customer
from core.address import Address

class TestOrderItem(unittest.TestCase):
    # Testing if Invoice 
    def test_creation(self):
        customer = Customer()
        address = Address("01420001")
        order = Order(customer, { "address": address })
        self.assertEqual(order.customer, customer)
        self.assertEqual(order.items, [])
        self.assertEqual(order.address, address)
    
    def test_add_product(self):
        product = Product("test name", "book")
        order = Order(Customer())
        # Adding product
        order.add_product(product)
        # Testing if product was added
        self.assertEqual(order.items[0].product, product)
    
    def test_total_amount(self):
        product1 = Product("test name 1", "book")
        product2 = Product("test name 2", "book")
        order = Order(Customer())
        order.add_product(product1)
        order.add_product(product2)
        # Testing total amount of two product is 20 (10 + 10)
        self.assertEqual(order.total_amount(), 20)

    def test_close(self):
        order = Order(Customer())
        # Closing order
        now_time = time.time()
        order.close(now_time)
        # Checking if it was closed
        self.assertEqual(order.closed_at, now_time)



if __name__ == "__main__":
    unittest.main()

File: backend/src/www/Tests/OrderTest.php
<?php
declare(strict_types=1);

use PHPUnit\Framework\TestCase;
use Creditas\Classes\Order;
use Creditas\Classes\Customer;
use Creditas\Classes\Product;

final class OrderTest extends TestCase
{
    public function testAddProduct()
    {
        $customer = new Customer("Juliana Apolo", "julianaapolo@gmail.com");
        $product = new Product("Livro", "book");
        $order = new Order($customer);
        $order->addProduct($product);
        $this->assertEquals('Livro', $order->items[0]->product->name);
        $this->assertEquals('book', $order->items[0]->product->type);
        return $order;
    }

    /**
     * @depends testAddProduct
     */
    public function testAddSecondProduct(Order $order)
    {
        $product = new Product("Assinatura Digital", "membership");
        $order->addProduct($product);

        $this->assertEquals(2, count($order->items));
        $this->assertEquals('Assinatura Digital', $order->items[1]->product->name);
        $this->assertEquals('membership', $order->items[1]->product->type);
    }

    /**
     * @depends testAddProduct
     */
    public function testTotalAmount(Order $order)
    {
        $this->assertEquals(20, $order->totalAmount());
    }

    /**
     * @depends testAddProduct
     */
    public function testClose(Order $order)
    {
        $time = time();
        $order->close($time);
        $this->assertEquals($time, $order->closedAt);
        $this->expectOutputString('<br>Shipping label: Juliana Apolo 45678-979 Item isento de impostos conforme disposto na Constituição Art. 150, VI, d.<br>Um e-mail com a mensagem \'Dados da sua assinatura...\' foi enviado para \'julianaapolo@gmail.com\'');
    }
}
